pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -1234,6 +1234,11 @@ protected void handleBrokerRegistrationEvent(String broker, NotificationType typ
             handleBrokerCreationEvent(broker);
         } else if (type == NotificationType.Deleted) {
             log.info("BrokerRegistry detected the broker:{} registry has been deleted.", broker);
+            // The registered node is an ephemeral node that could be deleted when the metadata store client's session
+            // is expired. In this case, we should register again.
+            if (brokerRegistry.getBrokerId().equals(broker)) {
+                brokerRegistry.registerAsync();
```

Comment:
Should we move this logic to `BrokerRegistry`?

Reply:
It makes sense. Moved now.